Kinvey/Kinvey/User.swift

```diff
@@ -584,8 +575,10 @@ open class User: NSObject, Credential, Mappable {
         let request = client.networkRequestFactory.buildUserSave(user: self, newPassword: newPassword)
         Promise<U> { fulfill, reject in
             request.execute() { (data, response, error) in
-                if let response = response , response.isOK, let user = client.responseParser.parseUser(data) as? U {
-                    client.activeUser = user
+                if let response = response, response.isOK, let user = client.responseParser.parseUser(data) as? U {
+                    if user.userId == client.activeUser?.userId {
```

Comment:
@heyzooi if this condition fails (the only reason I can imagine is because the active user changed while the `save` was in progress), then shouldn't we error out?